android_mobile/src/main/java/com/alexstyl/specialdates/addevent/ui/EventDatePicker.java

```diff
@@ -154,31 +152,15 @@ private int getYear() {
 
     private final NumberPicker.OnValueChangeListener dateValidator = new NumberPicker.OnValueChangeListener() {
 
-        private static final int LAST_DAY_OF_FEBRUARY_LONG = 29;
-        private static final int LAST_DAY_OF_FEBRUARY_SHORT = 28;
-
         @Override
         public void onValueChange(NumberPicker picker, int oldVal, int newVal) {
-            if (getMonth() == Months.FEBRUARY && isDisplayingYear()) {
-
-                if (isValidDate(LAST_DAY_OF_FEBRUARY_LONG, Months.FEBRUARY, getYear())) {
-                    dayPicker.setMaxValue(LAST_DAY_OF_FEBRUARY_LONG);
-                } else {
-                    dayPicker.setMaxValue(LAST_DAY_OF_FEBRUARY_SHORT);
-                }
+            if (isDisplayingYear()){
+                dayPicker.setMaxValue(Date.Companion.on(FIRST_DAY_OF_MONTH, getMonth(), getYear()).daysInCurrentMonth());
             } else {
-                dayPicker.setMaxValue(LAST_DAY_OF_MONTH);
+                dayPicker.setMaxValue(Date.Companion.on(FIRST_DAY_OF_MONTH, getMonth()).daysInCurrentMonth());
```

Comment:
I am a bit concerned here as this is going to be called every time the user scrolls the picker, right? 
On the other hand the `Date` object should be fairly simple to make, but yet it is work on the UI thread. 

An improvement would be to have a class thats spits out Date class according to the month (a map reference maybe?) 

It's just an improvement. Up to you to sort this out if you want to.

Reply:
Testing with profiler the computation is 64byte (if these that are counted are bytes) so I guess that any other solution will increase the memory footprint :wink:  

![profiler](https://user-images.githubusercontent.com/656323/39440680-5d3985d0-4cb4-11e8-8321-31447185b45e.PNG)